packages/form-builder/src/components/sidebar/secondary.js

```diff
@@ -1,12 +1,15 @@
-import {PanelHeader, createSlotFill} from '@wordpress/components';
-import {__experimentalListView} from '@wordpress/block-editor';
 import {__} from '@wordpress/i18n';
+import FieldTypesList from "./panels/field-types-list";
+import BlockListTree from "./panels/block-list-tree";
 
-const {Fill: InspectorFill} = createSlotFill(
-    'StandAloneBlockEditorSidebarInspector',
-);
+function Sidebar(props) {
 
-function Sidebar() {
+    const panels = {
+        add: () => <FieldTypesList />,
+        list: () => <BlockListTree />,
+    };
+
+    const PanelContent = panels[props.selected];
```

Comment:
Can you have this component receive the deconstructed props

`Sidebar({selected})` and specify the prop-type as well.

Reply:
@jonwaldstein does this look right? See https://github.com/impress-org/givewp-next-gen/pull/39/commits/e9ad8e34e7ad512fd7c0730be422edf57488da22

Reply:
```js
interface Props {
    selected: 'text'
}

const Sidebar: React.FC<Props> = ({selected}) => { ... }
```

Reply:
@kjohnson I would avoid using React.FC since it implies you also want to accept children.  For this I would just do

```
type PropTypes = {
    selected: string
}

const Sidebar = ({selected}: PropTypes) => {
```

or

```
const Sidebar = ({selected}: { selected: string }) => {
```

Types vs Interfaces are really interested & confusing in typescript.  I generally just use types for components props unless they are extending an existing interface of props or are intended to be extended.

Reply:
@jonwaldstein cool. Thanks for that. This is actually my first time using types with TypeScript.

Updated. See https://github.com/impress-org/givewp-next-gen/pull/39/commits/86b7b1a72f118bc3dfcf6366db908b1825b41dcf